src/stan/lang/ast_def.cpp

```diff
@@ -284,6 +284,16 @@ namespace stan {
     void function_signatures::add_unary(const::std::string& name) {
       add(name, DOUBLE_T, DOUBLE_T);
     }
+    void function_signatures::add_unary_vectorized(const::std::string&
+    name) {
+      for (size_t i = 0; i < 8; ++i) {
+        add(name, expr_type(DOUBLE_T, i), expr_type(INT_T, i));
```

Comment:
No
> On Sep 12, 2016, at 10:10 AM, Daniel Lee <notifications@github.com> wrote:
> 
> In src/stan/lang/ast_def.cpp:
> 
> > @@ -284,6 +284,16 @@ namespace stan {
> >      void function_signatures::add_unary(const::std::string& name) {
> >        add(name, DOUBLE_T, DOUBLE_T);
> >      }
> > +    void function_signatures::add_unary_vectorized(const::std::string&
> > +    name) {
> > +      for (size_t i = 0; i < 8; ++i) {
> > +        add(name, expr_type(DOUBLE_T, i), expr_type(INT_T, i));
> 
> this will promote all the function outputs for int inputs. Are there any cases where the output is int with an int input? It seems like there would be.

No.  Those'll be separate functions.    Only the double output
versions were vectorized.

With depth up to 7, that'll be 40 signatures per function.  The
parser should be able to handle that.

- Bob